instructure/canvas-ios
# Move http upload body creation from memory to file

refs: MBL-16104
affects: Student
release note: Fixed a crash on submitting big files.

test plan:
- Try to submit a big file with the share extension. (On an iPhone SE2 a 105 MB pdf was enough. The app also crashes but it requires a larger file.)
- Submission should go through without the share extension crashing.
- Check if file upload works from the application.

<table>
<tr><td/><th>Application memory usage while uploading a 105 MB pdf file</th></tr>
<tr>
<th>Before</th>
<td><img src="https://user-images.githubusercontent.com/72396990/177749296-4ec3d7d6-8405-46cd-8ffe-413eac9e0f0d.png"></td>
</tr>
<tr>
<th>After</th>
<td><img src="https://user-images.githubusercontent.com/72396990/177749336-52c066d5-478a-41bc-b558-f167dacb6f9b.png"></td>
</tr>
</table>
